src/main/java/taxi/service/authentication/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package taxi.service.authentication;
+
+import javax.naming.AuthenticationException;
+import taxi.dao.DriverDao;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverDao.findByLogin(login).orElseThrow(() ->
```

Comment:
```suggestion
        Optional<Driver> driver = driverDao.findByLogin(login);
```
Also you need to add in DriverService 'Optional<Driver> findByLogin(String login);'